assets/js/base/components/product-list/product-list.tsx

```diff
@@ -195,7 +207,7 @@ const ProductList = ( {
 		! Number.isFinite( totalProducts ) &&
 		Number.isFinite( previousQueryTotals?.totalProducts ) &&
 		isEqual( totalQuery, previousQueryTotals?.totalQuery )
-			? Math.ceil( previousQueryTotals.totalProducts / perPage )
+			? Math.ceil( ( previousQueryTotals?.totalProducts || 0 ) / perPage )
```

Comment:
Can we force this shape earlier to avoid checking for existance of totalProducts? Just for readability sake.

Reply:
The problem here is that `previousQueryTotals` could be `undefined`. I think this is the cleanest way using the optional chaining (`?.`) operator and defining a default value inline if the prop isn't defined. The alternative I see would be to wrap that entire block in an `if(previousQueryTotals) {}` statement. I don't think we can force `previousQueryTotals` to never be undefined. If you have any other ideas I'm open to suggestions